### ISSUE TITLE
Questions from Lecture on 11/10/2024

### ISSUE THREAD

**Issue by averageandyyy** [nus-cs2113-AY2425S1/forum#35]
Hi Prof, in the most recent lecture where we inferred that `ActionX` and `ActionY` were concrete classes and thus would have implemented both `perform()` and `repeat()`, their return types were not specified in their box. Should they have been `perform(): void` and `repeat(): void` instead?

Additionally, while I understand the dependency and reference between the `ActionFactory` and the `Action`s it would create, I am not confident that using a solid line to represent `ActionFactory`'s reference to `History` makes the most sense. One might argue that `History` can be a dependency in the sense that it is possible for us to pass `History` as a parameter to `ActionFactory` for the addition to occur. It is unclear, based on the instructions alone, that `ActionFactory` would contain a `History` instance. Would a dashed arrow indicating dependency in the form of a parameter be also an acceptable answer?

Thanks for the help!

**Comment by okkhoy** [comment]
> implemented both perform() and repeat(), their return types were not specified in their box. Should they have been perform(): void and repeat(): void instead?

yes. however, i think i mentioned in the lecture, i m skipping due to shortage of time (and space may be!)

> Would a dashed arrow indicating dependency in the form of a parameter be also an acceptable answer?

For the reasons you mentioned, yes.
I took the most common case where to call a method you need to "have**" a reference of the object whose method you call.

** here, have is interpreted as maintains a reference, hence solid line
